rviz_default_plugins/src/rviz_default_plugins/displays/image/ros_image_texture.cpp

```diff
@@ -294,6 +314,13 @@ ImageData ROSImageTexture::setFormatAndNormalizeDataIfNecessary(ImageData image_
       image_data.size_);
     image_data.pixel_format_ = Ogre::PF_BYTE_L;
     image_data.data_ptr_ = &buffer[0];
+  } else if ( // NOLINT enforces bracket on the same line, which makes code unreadable
+    image_data.encoding_ == sensor_msgs::image_encodings::YUV422 ||
+    image_data.encoding_ == sensor_msgs::image_encodings::YUV422_YUY2
+  )
+  {
+    // we previously converted from yuv to rgb so set the format to RGB
```

Comment:
Do the conversion here. Contain all our code inside those braces. Allocate a buffer for the conversion similar to  what they do in the condition above. The reallocate memory with every frame so if it's okay by them then it's okay by us. its unefficient but seems they do not care.

Reply:
They allocate a std::vector<uint8_t> buffer then point image_data.data_prt to it. You can do the same allocate buffer do the conversion from the msg to the buffer then make the image_data point to the buffer